### DISCUSSION TITLE
SC-Glow TTS paper for custom voices

### DISCUSSION THREAD

**Discussion by DwaraknathT** [coqui-ai/TTS#459] [General]
Hi everyone, 

I am not sure if I could ask this here, but the paper I am trying to work with uses coqui so I thought I'd give it a shot. I am trying to apply the code given [here](https://github.com/Edresson/SC-GlowTTS) to implement a TTS service with custom voice. From the given colab notebooks, It works really well on the given test samples.
However, when I try to use a sample from a custom dataset, the speaker embedding are not able to capture the features of the speaker. If I try to generate speech from a female voice reference, the generated voice is male.

I calculated the cosine similarities between reference and custom dataset speaker embeddings and here is what I found
Male - Female : 0.216
Male - Female custom : 0.028
Male - Male custom : -0.015
Female - Female custom: -0.02

Can you please let me know if there are any specific preprocessing steps that I to follow? the reference audio clips are Mono channel, 16KHz and clips of 10 seconds.

Thanks again for your work! Really appreciate it!

**Comment by erogol** [comment]
@Edresson might help you with that.

AFAIK, scglow does not really work well with a custom voice, at least for now, since it is only trained on VCTK dataset which is relatively small to generalize customer voices. I think @Edresson is planing to train with LIbriTTS to address this issue.

**Reply by Edresson** [reply]
@erogol Exactly. In the article, we train and compare the models using the VCTK dataset. As the VCTK is a set with a low number of speakers, the model is unable to generalize to all speakers and/or audio quality. The model can work or not depending on the characteristics of the speaker and audio.

 To achieve a better generalization for customized voices it is necessary to train the model using a larger amount of data and speakers (eg. LibriTTS / LibriSpeech) as done by [Jia et. al. 2018](https://arxiv.org/pdf/1806.04558.pdf) or in the repository [Real Time Voice Cloning](https://github.com/CorentinJ/Real-Time-Voice-Cloning) or in the works that apply zero-shot TTS models for ASR such as [Rosenberg et al. 2019](https://arxiv.org/pdf/1909.11699.pdf) and [Laptev et al. 2020](https://ieeexplore.ieee.org/document/9263564).

We showed in the article that SC-GlowTTS is able to learn the distribution of VCTK speakers with a smaller number of speakers. For this reason, in theory, we need fewer speakers than the Tacotron model to generalize well to custom speakers (subject to the limitations of the speaker verification model)

In the near future, I intend to train and make SC-GlowTTS available with the LibriTTS dataset.

**Reply by skol101** [reply]
@Edresson have you trained SC-GlowTTS with LibriTTS?

**Reply by Edresson** [reply]
> @Edresson have you trained SC-GlowTTS with LibriTTS?

@skol101  Hi, I haven't trained yet. But we have better results than SC-GlowTTS and we will make checkpoints available soon (we are writing the paper).

**Reply by skol101** [reply]
@Edresson cheers!
